src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/policy/Input.kt

```diff
@@ -1,17 +1,82 @@
 package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy
 
+import com.fasterxml.jackson.annotation.JsonValue
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalConditionData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.InputType.ADDRESS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.InputType.TEXT
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.Case
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.adjustCase
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.formatAddress
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.formatUsing
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.startsWithVowel
+
+enum class InputType(@JsonValue val description: String) {
+  RADIO("radio"),
+  ADDRESS("address"),
+  TIME_PICKER("timePicker"),
+  DATE_PICKER("datePicker"),
+  FILE_UPLOAD("fileUpload"),
+  TEXT("text"),
+  CHECK("check"),
+}
+
 data class Input(
-  val type: String,
+  override val type: InputType,
   val label: String,
   val name: String,
-  val listType: String? = null,
+  override val listType: String? = null,
   val options: List<Option>? = null,
-  val case: String? = null,
-  val handleIndefiniteArticle: Boolean? = null,
+  override val case: Case? = null,
+  override val handleIndefiniteArticle: Boolean? = null,
   val addAnother: AddAnother? = null,
-  val includeBefore: String? = null,
+  override val includeBefore: String? = null,
   val subtext: String? = null,
-)
+) : FormattingRule
+
+interface FormattingRule {
+  val type: InputType
+  val case: Case?
+  val listType: String?
+  val includeBefore: String?
+  val handleIndefiniteArticle: Boolean?
+
+  companion object {
+    val DEFAULT = object : FormattingRule {
+      override val type = TEXT
+      override val case = null
+      override val listType = "AND"
+      override val includeBefore = null
+      override val handleIndefiniteArticle = null
+    }
+  }
+}
+
+fun FormattingRule.format(data: List<AdditionalConditionData>) = when (data.size) {
+  0 -> ""
+  1 -> formatSingleValue(data.first().dataValue!!)
+  else -> formatMultipleValues(data)
+}
+
+private fun FormattingRule.formatSingleValue(item: String) = item
+  .adjustCase(this.case)
+  .let { this.applyFormattingRules(it) }
+
+private fun FormattingRule.formatMultipleValues(data: List<AdditionalConditionData>) = data
+  .mapNotNull { it.dataValue }
+  .map { it.adjustCase(this.case) }
```

Comment:
this is slightly different logic to the frontend - this would previously create the joined string of items and then apply casing rules which doesn't make sense for the joining word:

before: 
`[an, bn ,cn]` => `an, bn and cn` => `An, Bn, And Cn`
 
after:
`[an, bn ,cn]` => `an, bn and cn` => `An, Bn, and Cn`

Reply:
got this confirmed by the designers